src/spatio_temporal_voxel_layer.cpp

```diff
@@ -545,12 +545,11 @@ void SpatioTemporalVoxelLayer::activate(void)
   RCLCPP_INFO(logger_, "%s was activated.", getName().c_str());
 
   observation_subscribers_iter sub_it = _observation_subscribers.begin();
-  for (; sub_it != _observation_subscribers.end(); ++sub_it) {
-    (*sub_it)->subscribe();
-  }
-
   observation_buffers_iter buf_it = _observation_buffers.begin();
-  for (; buf_it != _observation_buffers.end(); ++buf_it) {
+  for (; sub_it != _observation_subscribers.end(); ++sub_it, ++buf_it) {
+    if (!(*buf_it)->IsEnabled()) {
```

Comment:
buf it is not set anywhere of anything except `begin()`

Reply:
it is set from the `toggle` service [callback ](https://github.com/SteveMacenski/spatio_temporal_voxel_layer/blob/94e8439f7d1ff7038ce9c1138df5d2e7c743dbc2/src/spatio_temporal_voxel_layer.cpp#L444)
I think this was initially the desired functionality, the `enable` parameter can disable the subscription on startup, and then it can subscribe by calling the service. [docstring](https://github.com/SteveMacenski/spatio_temporal_voxel_layer/blob/94e8439f7d1ff7038ce9c1138df5d2e7c743dbc2/src/spatio_temporal_voxel_layer.cpp#L238)

Reply:
The orders of sub_it and buf_its are not guaranteed to be aligned if the configuration is malformed.